src/interpolate/Interpolate_unit.re

```diff
@@ -0,0 +1,33 @@
+type cssUnit = {
+  num: float,
+  unit: Js.Nullable.t(string),
+};
+
+[@bs.deriving jsConverter]
+type measurement = [ | `px | `em | `rem | `vw | `vh | [@bs.as "%"] `pct];
```

Comment:
I talked about this a bit here: https://formidablelabs.slack.com/archives/CR02GE56Y/p1576533160023200. Basically, when we get a string from JS-land we call `measurementFromJs`, which will attempt to convert it to one of these polymorphic variants. If it finds a match, we'll get `Some`, otherwise `None`. Then on L16-17 below we convert it back. This is what allows us to return `null` for `unit` if the passed `unit` doesn't match one of our polymorphic variants. Happy to go more in depth on this if folks want. Here's the associated documentation: https://bucklescript.github.io/docs/en/generate-converters-accessors#convert-between-js-string-enum-and-bs-polymorphic-variant